src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.order
+
+/-!
+# Lexicographic order on finitely supported functions
+
+This file defines the lexicographic order on `finsupp`.
+-/
+
+variables {α N : Type*}
+
+namespace finsupp
+
+lemma filter_ne_eq_empty_iff [decidable_eq α] [decidable_eq N] [has_zero N] {f g : α →₀ N} :
+  (f.support ∪ g.support).filter (λ a, f a ≠ g a) = ∅ ↔ f = g :=
+begin
```

Comment:
You are probably right.  The lemma seems to be [docs#finset.filter_eq_empty_iff](https://leanprover-community.github.io/mathlib_docs/data/finset/basic.html#finset.filter_eq_empty_iff) and is not a simp lemma, it seems.  I'll try it when I'm back at a computer, thanks!

Reply:
Ah that explains it! Sorry for the confusing comment, hope it is still useful ☺️

Reply:
I have managed to find a proof going via the lemma that you suggested, but it is essentially the same length as the one that I had.
```lean
lemma filter_ne_eq_empty_iff [decidable_eq α] [decidable_eq N] [has_zero N] {f g : α →₀ N} :
  (f.support ∪ g.support).filter (λ a, f a ≠ g a) = ∅ ↔ f = g :=
begin
  simp only [finset.filter_eq_empty_iff, finset.mem_union, mem_support_iff, not_not],
  refine ⟨λ h, ext (λ a, _), λ h x hx, congr_fun h x⟩,
  by_cases fg : ¬f a = 0 ∨ ¬g a = 0,
  { exact h _ fg },
  { rw [← not_and_distrib, not_not] at fg,
    simp [fg] }
end
```

I'm keeping the current version, just because it is already here!